app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -498,62 +503,118 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
+                    // TODO isn't this condition always true, shouldn't it be & instead of | ?
+                    mBluetoothGatt.setCharacteristicNotification(gattCharacteristic, true);
 
-                try {
-                    final BluetoothGattDescriptor bdescriptor = gattCharacteristic.getDescriptor(UUID.fromString(HM10Attributes.CLIENT_CHARACTERISTIC_CONFIG));
-                    Log.i(TAG, "Bluetooth Notification Descriptor found: " + bdescriptor.getUuid());
-                    bdescriptor.setValue(BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE);
-                    mBluetoothGatt.writeDescriptor(bdescriptor);
-                } catch (Exception e) {
-                    Log.e(TAG, "Error setting notification value descriptor: " + e);
-                }
+                    try {
+                        final BluetoothGattDescriptor bdescriptor = gattCharacteristic.getDescriptor(UUID.fromString(HM10Attributes.CLIENT_CHARACTERISTIC_CONFIG));
+                        Log.i(TAG, "Bluetooth Notification Descriptor found: " + bdescriptor.getUuid());
+                        bdescriptor.setValue(BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE);
+                        mBluetoothGatt.writeDescriptor(bdescriptor);
+                    } catch (Exception e) {
+                        Log.e(TAG, "Error setting notification value descriptor: " + e);
+                    }
 
-                // Experimental support for "Transmiter PL" from Marek Macner @FPV-UAV
-                //if (use_transmiter_pl_bluetooth) {
+                    // Experimental support for "Transmiter PL" from Marek Macner @FPV-UAV
+                    //if (use_transmiter_pl_bluetooth) {
                     // BluetoothGattDescriptor descriptor = gattCharacteristic.getDescriptor(UUID.fromString(HM10Attributes.CLIENT_CHARACTERISTIC_CONFIG));
                     //   Log.i(TAG, "Transmiter Descriptor found: " + descriptor.getUuid());
                     //   descriptor.setValue(BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE);
                     //   mBluetoothGatt.writeDescriptor(descriptor);
-                //}
-
-                // Experimental support for rfduino from Tomasz Stachowicz
-                if (use_rfduino_bluetooth) {
-                    //  BluetoothGattDescriptor descriptor = gattCharacteristic.getDescriptor(UUID.fromString(HM10Attributes.CLIENT_CHARACTERISTIC_CONFIG));
-                    //  Log.i(TAG, "Transmiter Descriptor found use_rfduino_bluetooth: " + descriptor.getUuid());
-                    //  descriptor.setValue(BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE);
-                    //  mBluetoothGatt.writeDescriptor(descriptor);
-                    Log.w(TAG, "onServicesDiscovered: use_rfduino_bluetooth send characteristic " + xDripDataCharacteristicSend + " found");
-                    mCharacteristicSend = gattService.getCharacteristic(xDripDataCharacteristicSend);
+                    //}
```

Comment:
I'm confused from the diff exactly what has changed here, could you briefly explain the code flow? Thanks